public/controllers/agent/register-agent/steps/wz-manager-address.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { memo, useCallback, useEffect, useState } from 'react';
+import { EuiText, EuiFieldText } from '@elastic/eui';
+
+type Props = {
+  onChange: (value: string) => void;
+  defaultValue?: string;
+};
+
+const WzManagerAddressInput = (props: Props) => {
+  const { onChange, defaultValue } = props;
+  const [value, setValue] = useState('');
+
+  useEffect(() => {
+    if(defaultValue){
+      setValue(defaultValue);
+      onChange(defaultValue);
+    }else{
+      setValue('');
+      onChange('');
+    }
+  },[])
+  /**
+   * Handles the change of the selected node IP
+   * @param value
+   */
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    onChange(value);
+    setValue(value);
+  };
+  return (
+    <EuiText>
+      <p>
+        This is the address the agent uses to communicate with the Wazuh server.
+        It can be an IP address or a fully qualified domain name (FQDN).
+      </p>
+      <EuiFieldText
+        placeholder='Server Address'
+        onChange={handleOnChange}
+        value={value}
+      />
+    </EuiText>
+  );
+};
+
+export default memo(WzManagerAddressInput);
```

Comment:
**thought**: what is the reason to use `memo` here? Could you explain?

Reply:
I removed the memo. I was tried to prevent unnecessary re-renders of server address component but is not possible because of the parent component implementation. The server address component is always re-rendered because is defined inside the parent component state. The cost of prevent re-renders is high, we must refactor the register-agent component to make it.